Proteomics/Protein/Protein.cs

```diff
@@ -275,6 +278,73 @@ internal static IEnumerable<PeptideWithSetModifications> RemoveTerminalModificat
             }
         }
 
+        internal IEnumerable<PeptideWithSetModifications> GetCleavageModifiedPeptides(IEnumerable<PeptideWithSetModifications> originalPeptides, DigestionParams digestionParams)
+        {
+            if (digestionParams.Protease.CleavageMassShifts.Count ==0)
+            {
+                foreach(var peptide in originalPeptides)
+                {
+                    yield return peptide;    
+                }                
+            }
+            else
+            {
+                Dictionary<string, (int, double)> residueSpecifics = new Dictionary<string, (int, double)>();
+                var massShifts = digestionParams.Protease.CleavageMassShifts;
+                var digestionMotifs = digestionParams.Protease.DigestionMotifs;
+                foreach (var motif in digestionMotifs)
+                {
+                    var cleavageResidue = motif.InducingCleavage;
+                    // cutIndex of 0 is N and 1 is C
+                    var cleavageIndex = motif.CutIndex;
+                    if (massShifts.ContainsKey(cleavageResidue))
+                    {
+                        residueSpecifics.Add(cleavageResidue, (cleavageIndex, massShifts[cleavageResidue]));
```

Comment:
would this work if 'X' was the motif residue? if you implemented this as a mod you could just use `ModificationLocalization.ModFits` to determine if the mod (CNBr mass shift) fits at this location.

actually on second thought I think all you'd have to do is in the `Protein.Digest` method, just add the mod that comes from CNBr cleavage to fixed mods. I think none of the subsequent code would have to change?